src/tiled/mapsdock.cpp

```diff
@@ -41,6 +41,26 @@
 using namespace Tiled;
 using namespace Tiled::Internal;
 
+/**
+ * Class represents the file system model with disabled dragging of directories.
+ */
+class FileSystemModel: public QFileSystemModel
+{
+public:
+    explicit FileSystemModel(QObject *parent = 0):
+        QFileSystemModel(parent)
+    {
+    }
+
+    virtual Qt::ItemFlags flags(const QModelIndex &i) const
```

Comment:
Ah, instead of repeating the `virtual` keyword, which does nothing because the method is already virtual, it would be more useful to mark this method at the end with `override`, which makes the compiler verify that it's overriding a virtual method.